From ce36c45f4509f893622cd239911a633e5c279ac5 Mon Sep 17 00:00:00 2001
From: Kevin Hao <kexin.hao@windriver.com>
Date: Thu, 22 Oct 2009 10:37:33 +0800
Subject: [PATCH 60/77] net/dpa: disconnect phy when stopping net dev

The dpa driver doesn't disconnect phy when stoping net dev.
This will cause a phy reattach error when we down the device and
up it again.

Signed-off-by: Kevin Hao <kexin.hao@windriver.com>
---
 drivers/net/dpa/dpa.c |    2 ++
 1 files changed, 2 insertions(+), 0 deletions(-)

diff --git a/drivers/net/dpa/dpa.c b/drivers/net/dpa/dpa.c
index 553ca5d..2156152 100644
--- a/drivers/net/dpa/dpa.c
+++ b/drivers/net/dpa/dpa.c
@@ -1742,6 +1742,8 @@ static int __cold dpa_stop(struct net_device *net_dev)
 
 		for (i = 0; i < ARRAY_SIZE(priv->mac_dev->port_dev); i++)
 			fm_port_disable(priv->mac_dev->port_dev[i]);
+		phy_disconnect(priv->mac_dev->phy_dev);
+		priv->mac_dev->phy_dev = NULL;
 	}
 
 	if (netif_msg_ifdown(priv))
-- 
1.6.5.2

